packages/fields/src/types/Slug/Implementation.js

```diff
@@ -0,0 +1,258 @@
+import slugify from '@sindresorhus/slugify';
+import cuid from 'cuid';
+import {
+  Text,
+  MongoTextInterface as MongoSlugInterface,
+  KnexTextInterface as KnexSlugInterface,
+} from '../Text/Implementation';
+
+const MAX_UNIQUE_ATTEMPTS = 100;
+
+const findFirstNonEmptyStringValue = fields =>
+  Object.values(fields).find(value => typeof value === 'string' && value);
+
+const generateSlug = valueToSlugify => slugify(valueToSlugify || '');
+
+export class SlugImplementation extends Text {
+  constructor(
+    path,
+    { from, generate, makeUnique, isUnique, regenerateOnUpdate = true },
+    { listKey }
+  ) {
+    const listAndFieldPath = `${listKey}.${path}`;
+
+    let generateFn = generate;
+    let makeUniqueFn = makeUnique;
+
+    if (typeof regenerateOnUpdate !== 'boolean') {
+      throw new Error(`The 'regenerateOnUpdate' option on ${listAndFieldPath} must be true/false`);
+    }
+
+    if (from && generate) {
+      throw new Error(
+        `Only one of 'from' or 'generate' can be supplied as an option to the Slug field on ${listAndFieldPath}.`
+      );
+    }
+
+    if (from) {
+      if (typeof from !== 'string') {
+        if (typeof from === 'function') {
```

Comment:
I like this use of nested if over `&&` :)

Reply:
Honestly, without it, I was going around in loops in my head trying to keep a handle on all the different branches!